app/src/main/java/com/lumisdinos/gameoffifteen/domain/model/GameModel.kt

```diff
@@ -0,0 +1,7 @@
+package com.lumisdinos.gameoffifteen.domain.model
+
+data class GameModel(
+    var id: Int = 0,
```

Comment:
never use variables inside of models. val instead var !

Reply:
I set id as val, but other fields I need as vars because they may be changed

Reply:
you have to use a model instance(new exemplar), not this model concretely

Reply:
Data class is immutable, you should not use "var" inside. Use copy() If you need to change something inside data class.

Reply:
done